test/e2e_node/plugins/gcp-credential-provider/README.md

```diff
@@ -0,0 +1,35 @@
+# GCP credential provider for e2e testing
+
+This package contains a barebones implementation of the [kubelet GCP credential
```

Comment:
> barebones implementation

how much different is it from the real one? What is blocking from using a real one? My worry as with any fake that we may end up chasing issues that would result from the infra and the real provider updates.

Reply:
> how much different is it from the real one?

There are quite a few differences, the actual implementation is here: https://github.com/kubernetes/cloud-provider-gcp/tree/master/cmd/auth-provider-gcp 

> What is blocking from using a real one?

Having a local copy just for testing is really to avoid dependencies we would have with k8s.io/cloud-provider-gcp. We're trying to remove go module dependencies to cloud provider implementations eventually so it would be counter intuitive to import k8s.io/cloud-provider-gcp in order to test the credential provider mechanism. Luckily the GCP provider doesn't require vendoring a cloud SDK so a local copy just for node e2e testing is actually do-able.   

> My worry as with any fake that we may end up chasing issues that would result from the infra and the real provider updates.

This is fair and I think the full e2e test suite (not node e2e) should use the actual provider configured via prow jobs. This is something @adisky was working on. The goal of this specific PR is to provide test coverage for the kubelet plugin mechanism itself, it just so happens that we run node e2es on GCE nodes and all the authenticated images are on gcr.io which necessitates this.  

I think we're trying to find a balance here of doing the bare minimum to only validate the core Kubernetes parts, i.e. the kubelet code executing the plugins and not the plugins themselves, which is difficult because we heavily rely on GCP infrastructure for our tests.

Reply:
ok, it is reasonable then. If there is an opportunity to shrink code even further, it will be good.